src/components/step-tooltip/index.js

```diff
@@ -0,0 +1,169 @@
+import React, { memo, useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  HeadingText,
+  Link,
+  Modal,
+  Popover,
+  PopoverBody,
+  PopoverTrigger,
+} from 'nr1';
+import IconsLib from '../icons-lib';
+import { SIGNAL_TYPES, UI_CONTENT } from '../../constants';
+
+const StepToolTip = memo(
+  ({
+    stepTitle,
+    stepStatus,
+    stepStatusOption,
+    stepStatusUnit,
+    stepStatusValue,
+    signals,
+    triggerElement,
+  }) => {
+    const [includedSignals, setIncludedSignals] = useState([]);
+    const [viewRuleModalHidden, setViewRuleModalHidden] = useState(true);
+
+    useEffect(() => {
+      if (signals.length > 0) {
+        const includedSignals = signals.filter((s) => {
+          return s.included === true || s.included === undefined;
+        });
+
+        setIncludedSignals(includedSignals);
+      }
```

Comment:
I would change this to...
```javascript
if (signals.length > 0) {
  setIncludedSignals(() => signals.filter((s) => {
    return s.included === true || s.included === undefined;
  }));
```

`includedSignals` is already defined as a state variable, and to remove ambiguity, I'd try not creating a local variable named `includedSignals`